airbyte-integrations/bases/source-test-lib/src/main/java/io/airbyte/integrations/base/TestSource.java

```diff
@@ -0,0 +1,265 @@
+   * postgres database. This function will be called before EACH test.
+   *
+   * @param testEnv - information about the test environment.
+   * @throws Exception - can throw any exception, test framework will handle.
+   */
+  protected abstract void setup(TestDestinationEnv testEnv) throws Exception;
+
+  /**
+   * Function that performs any clean up of external resources required for the test. e.g. delete a
+   * postgres database. This function will be called after EACH test. It MUST remove all data in the
+   * destination so that there is no contamination across tests.
+   *
+   * @param testEnv - information about the test environment.
+   * @throws Exception - can throw any exception, test framework will handle.
+   */
+  protected abstract void tearDown(TestDestinationEnv testEnv) throws Exception;
+
+  @Before
+  public void setUpInternal() throws Exception {
+    Path testDir = Path.of("/tmp/airbyte_tests/");
+    Files.createDirectories(testDir);
+    final Path workspaceRoot = Files.createTempDirectory(testDir, "test");
+    jobRoot = Files.createDirectories(Path.of(workspaceRoot.toString(), "job"));
+    localRoot = Files.createTempDirectory(testDir, "output");
+    testEnv = new TestDestinationEnv(localRoot);
+
+    setup(testEnv);
+
+    pbf = new DockerProcessBuilderFactory(
+        workspaceRoot,
+        workspaceRoot.toString(),
+        localRoot.toString(),
+        "host");
+  }
+
+  @After
+  public void tearDownInternal() throws Exception {
+    tearDown(testEnv);
+  }
+
+  /**
+   * Verify that when the integrations returns a valid spec.
+   */
+  @Test
+  public void testGetSpec() {
+    final OutputAndStatus<StandardGetSpecOutput> output = runSpec();
+    assertTrue(output.getOutput().isPresent());
+  }
+
+  /**
+   * Verify that when given valid credentials, that check connection returns a success response.
+   * Assume that the {@link TestSource#getConfig()} is valid.
+   */
+  @Test
+  public void testCheckConnection() throws Exception {
+    final OutputAndStatus<StandardCheckConnectionOutput> output = runCheck();
+    assertTrue(output.getOutput().isPresent());
+    assertEquals(Status.SUCCEEDED, output.getOutput().get().getStatus());
+  }
+
+  // /**
+  // * Verify that when given invalid credentials, that check connection returns a failed response.
+  // * Assume that the {@link TestSource#getFailCheckConfig()} is invalid.
+  // */
+  // @Test
+  // public void testCheckConnectionInvalidCredentials() throws Exception {
+  // final OutputAndStatus<StandardCheckConnectionOutput> output = runCheck();
+  // assertTrue(output.getOutput().isPresent());
+  // assertEquals(Status.FAILED, output.getOutput().get().getStatus());
+  // }
+
+  /**
+   * Verify that when given valid credentials, that discover returns a valid catalog. Assume that the
+   * {@link TestSource#getConfig()} is valid.
+   */
+  @Test
+  public void testDiscover() throws Exception {
```

Comment:
Should we also validate that this matches the catalog we expect?

Reply:
legit. artifact of when i wasn't passing the catalog in, but now that we are, you're right.